src/usecase_clustering.Rmd

```diff
@@ -0,0 +1,148 @@
+data(agriculture)
+
+plot(y ~ x, data = agriculture)
+```
+
+
+We aim to group the observations into clusters that contain similar objects. We will 
+
+* define the learning task  ([here](http://mlr-org.github.io/mlr-tutorial/devel/html/task/index.html)),
+* select a learning method ([here](http://mlr-org.github.io/mlr-tutorial/devel/html/learner/index.html)),
+* train the learner with data ([here](http://mlr-org.github.io/mlr-tutorial/devel/html/train/index.html)), 
+* evaluate the performance of the model ([here](http://mlr-org.github.io/mlr-tutorial/devel/html/performance/index.html)) and
+* tune the model ([here](http://mlr-org.github.io/mlr-tutorial/devel/html/tune/index.html)).
+
+### Defining a task
+
+We now have to define a clustering task. Notice that a clustering task doesn't have a target variable. 
+
+```{r message = FALSE}
+library(mlr)
+agri.task = makeClusterTask(data = agriculture)
+agri.task
+```
+
+Calling the task again shows us some basic informations as the number of observations, the data types of the features or if there are still some missing values, that should have been preprocessed.
+
+### Defining a learner
+
+We generate the learner by calling ``makeLearner`` and specifying the learning method, and, if needed, hyperparameters.
+
+An overview over all learners can be found  [here](http://mlr-org.github.io/mlr-tutorial/devel/html/integrated_learners/index.html). You can also call the \texttt{listLearners} command for our specific task.
+
+
+```{r, warning=FALSE, eval = FALSE}
+listLearners(obj = agri.task)
+```
+
+We will apply the $k$-means algorithm with $3$ centers for the moment
+
+```{r}
+cluster.lrn = makeLearner("cluster.kmeans", centers = 3)
+cluster.lrn
+```
+### Train the model
+
+The next step is to train our learner by feeding it with our data.
+
+```{r}
+agri.mod = train(learner = cluster.lrn, task = agri.task)
+```
+
+We can extract the model and have a look at it. 
+
+```{r}
+getLearnerModel(agri.mod)
+```
+
+### Prediction
+
+Now, we can predict the target values, our cluster labels.  
+
+```{r}
+agri.pred = predict(agri.mod, task = agri.task)
+agri.pred
+```
+
+### Performance 
+
+Since the data given to the learner is unlabeled, there is no objective evaluation of the accuracy of our model. We have to consider other criterions in unsupervised learning. 
+
+An overview over all performance measures can be found  [here](http://mlr-org.github.io/mlr-tutorial/devel/html/measures/index.html). You can also call the \texttt{listMeasures} command for our specific task.
+
+```{r}
+listMeasures(agri.task)
+```
+
+Let's have a look at the silhouette coefficient and the Davies-Boulding index.
+
+```{r message = FALSE}
+library("clValid")
+
+performance(agri.pred, measures = list(silhouette, db), task = agri.task)
+```
+
+### Tuning
+
+It's hard to say if our clustering is good since up to now we have nothing to compare to. Could we have done better by choosing another parameter for the number of centers? 
```

Comment:
I would replace "choosing another parameter for the number of centers" by "choosing a different number of centers"